source/opt/ir_context.h

```diff
@@ -394,10 +405,20 @@ class IRContext {
   opt::DominatorAnalysis* GetDominatorAnalysis(const ir::Function* f,
                                                const ir::CFG&);
 
+  // Gets the dominator analysis for function |f|.
+  opt::DominatorAnalysis* GetDominatorAnalysis(const ir::Function* f) {
+    return GetDominatorAnalysis(f, *cfg());
```

Comment:
Yes, I think it was a misunderstanding on how the whole thing worked when it was first introduced. I didn't touched the interface to avoid any breakage (but I think it needs a clean up).

Are you fine if I address that in a specific PR to avoid to much noise ?